src/antsibull_docs/collection_config.py

```diff
@@ -50,10 +50,10 @@ async def load_collection_config(
         config_path = os.path.join(collection_path, "docs", "docsite", "config.yml")
         if os.path.isfile(config_path):
             try:
-                return CollectionConfig.parse_obj(load_yaml_file(config_path))
-            except v1.ValidationError:
+                return CollectionConfig.model_validate(load_yaml_file(config_path))
+            except p.ValidationError:
                 pass
-        return CollectionConfig.parse_obj({})
+        return CollectionConfig.model_validate({})
```

Comment:
```suggestion
        return CollectionConfig()
```

should be equivalent, but more efficient